app/controllers/results_controller.rb

```diff
@@ -138,12 +138,11 @@ def edit
 
   def run_tests
     grouping_id = params[:grouping_id]
-    # TODO: The submission id is set incorrectly as the grouping_id
     submission_id = Result.find(params[:id]).submission.id
 
     begin
       AutomatedTestsHelper.request_a_test_run(grouping_id,
-                                              'submission',
+                                              'request',
```

Comment:
The "run tests" button is meant to be pushed by TAs and Admins, not students. (If the button shows for students that's a bug. They should only see that whole page once the assignment has been marked and released.)